### PR TITLE
Fix Advanced Autodiff docs notebook

### LINKED ISSUE #18219: Constant factor error in JAX 101 docs
### Description

In the [Advanced Autodiff](https://jax.readthedocs.io/en/latest/jax-101/04-advanced-autodiff.html) section of the JAX 101 docs, there is a bug in the TD(0) section: the **pseudo loss function** is wrong and needs a constant factor of -0.5 throughout this section.

I have implemented a fix for this [here](https://github.com/google/jax/pull/18220) and added an extra code cell to confirm the fix is correct. 

### What jax/jaxlib version are you using?

jaxlib v0.4.19

### Which accelerator(s) are you using?

N/A

### Additional system info

N/A

### NVIDIA GPU info

_No response_

### PR DESCRIPTION
**Pull request by ArthurConmy**
Fixes https://github.com/google/jax/issues/18219

Also, fixes the `stop_gradient` part of https://github.com/google/jax/issues/5935 as I implement a cell which tests that the constant factor is correct, and doesn't use `stop_gradient`.

### CODE REVIEW

**Comment by mattjj** [comment]
Thank you!

**Review by mattjj — APPROVED** [review]

**Comment by mattjj** [comment]
Unfortunately there's an issue with jupytext. It may be fixable by just running `pip install --upgrade jupytext` and then `jupytext --sync docs/jax-101/04-advanced-autodiff.md`.

@ArthurConmy could you try running those commands and pushing to your branch?

**Comment by jakevdp** [comment]
More info here: https://jax.readthedocs.io/en/latest/developer.html#syncing-notebooks

The easiest way to make this work is to run
```
git add . -u
pre-commit run jupytext
```
assuming you've installed `pre-commit` as mentioned on that page.

**Comment by hawkinsp** [comment]
I merged this via #18615 (I left the commit attributed to you unchanged), all I did was fix the presubmit problem in a second commit.
